appgate/customloaders.py

```diff
@@ -24,12 +24,38 @@ class CustomLoader:
 class CustomAttribLoader(CustomLoader):
     loader: Callable[[Any], Any] = attrib()
     field: str = attrib()
+    load_external: bool = attrib()
+
+    def load(self, values: AttributesDict) -> AttributesDict:
+        if self.load_external:
+            # Try loading the attribute's value from the external source
+            v = self.loader(values)
+            if not v:
+                return values
+            values[self.field] = v
+        else:
+            # If internal source or k8s secret method, see if the attribute is in the values
+            v = values.get(self.field)
+            if not v:
+                return values
+            values[self.field] = self.loader(v)
+        return values
+
+
+@attrs()
+class FileAttribLoader(CustomAttribLoader):
+    loader: Callable[[Any], Any] = attrib()
+    field: str = attrib()
+    load_external: bool = attrib()
 
     def load(self, values: AttributesDict) -> AttributesDict:
         v = values.get(self.field)
-        if not v:
-            return values
-        values[self.field] = self.loader(v)
+        if not v and self.load_external:
+            # Try loading the s value from the external source if value doesn't exist
+            v = self.loader(values)
```

Comment:
Same, what happens if the loader raises an exception here?

Reply:
Same as the exception thrown in CustomAttribLoader